File: src/dataset/Java/rna-transcription_exercism-java-ThomasZumsteg/claude/ClaudeSonnet4_rna-transcription_v2.java
public class RnaTranscription {
    public static String ofDna(String dna) {
        StringBuilder rna = new StringBuilder(dna.length());
        for (int i = 0; i < dna.length(); i++) {
            char c = dna.charAt(i);
            switch(c) {
                case 'C': rna.append('G'); break;
                case 'G': rna.append('C'); break;
                case 'T': rna.append('A'); break;
                case 'A': rna.append('U'); break;
            }
        }
        return rna.toString();
    }
}

File: src/dataset/c/rna_transcription_Exercism_ThomasZumsteg/claude/ClaudeSonnet4_rna_transcription_v2.c
#include "rna_transcription.h"
#include <stdlib.h>
#include <string.h>

static const char rna_map[256] = {
    ['G'] = 'C',
    ['T'] = 'A', 
    ['A'] = 'U',
    ['C'] = 'G'
};

char *to_rna(const char *dna) {
    size_t len = strlen(dna);
    char *rna = malloc(len + 1);
    if (!rna) return NULL;
    
    for (size_t i = 0; i < len; i++) {
        char mapped = rna_map[(unsigned char)dna[i]];
        if (!mapped) {
            free(rna);
            return NULL;
        }
        rna[i] = mapped;
    }
    rna[len] = '\0';
    return rna;
}